service/matching/task_queue_manager.go

```diff
@@ -459,7 +389,7 @@ func (c *taskQueueManagerImpl) AddTask(
 	// specific queues could cause them to get stuck behind "compatible" tasks when they should be able to progress
 	// independently.
 	if taskInfo.VersionDirective.GetUseDefault() != nil {
-		err = params.baseTqm.SpoolTask(params)
+		err = c.SpoolTask(params)
 	} else {
 		err = c.SpoolTask(params)
 	}
```

Comment:
simplify?
and change the comment above?

but is this behavior changing now, though? it seems like we should only do it if using the new api.